dfastmi/batch/ConfigurationInitializer.py

```diff
@@ -0,0 +1,309 @@
+        self._set_fraction_times(reach, self.q_threshold, self.discharges)
+
+        # determine the bed celerity based on the input settings
+        self._celerity = self.get_bed_celerity(reach, self.discharges)
+
+        self._rsigma = relax_factors(self.discharges, self.time_fractions_of_the_year, reach.qstagnant, self.celerity, reach.normal_width)
+        self._tstag = 0.0
+
+        self._n_fields = self._get_tide(reach, config)
+        self._needs_tide = reach.use_tide
+        if self.needs_tide:
+            self._tide_bc = reach.tide_boundary_condition
+
+    def _get_tide(self, reach: Reach, config: ConfigParser) -> int:
+        """
+        When using tide retrieve from configuration else return default 1
+
+        Arguments
+        ---------
+        reach : Reach
+            The reach we want to get the levels from.
+        config : ConfigParser
+            Configuration of the analysis to be run.
+        
+        Return
+        ------
+        n_fields : int
+            An int stating the number of fields
+        """
+        
+        if reach.use_tide:
+            try:
+                n_fields = int(config.get("General", "NFields", fallback=""))
+            except ValueError:
+                n_fields = 1 # Or should I raise ValueError exception?
+            if n_fields == 1:
+                raise ValueError("Unexpected combination of tides and NFields = 1!")
+        else:
+            n_fields = 1
+        return n_fields
+
+    @staticmethod
+    def get_bed_celerity(reach : Reach, discharges :Vector) -> Vector:
+        """
+        Will create a vector of values each representing the bed celerity for the 
+        period given by the corresponding entry in discharge (Q) [m/s] by the 
+        celerity type (via a discharge or via legacy using properties).
+
+        Arguments
+        ---------
+        reach : Reach
+            The reach we want to get the levels from.
+        discharges : Vector
+            A vector of discharges (Q) included in hydrograph [m3/s].
+        
+        Return
+        ------
+        celerity : Vector
+            A vector of values each representing the bed celerity for the 
+            period given by the corresponding entry in Q [m/s].      
+        """
+        cform = reach.celer_form
+        celerity = ()
+        if cform == 1:
+            prop_q = reach.celer_object.prop_q
+            prop_c = reach.celer_object.prop_c
+            celerity = tuple(get_celerity(q, prop_q, prop_c) for q in discharges)
+        elif cform == 2:
+            cdisch = reach.celer_object.cdisch
+            celerity = tuple(cdisch[0]*pow(q,cdisch[1]) for q in discharges)
+
+         # set the celerity equal to 0 for discharges less or equal to qstagnant
+        celerity = tuple({False:0.0, True:celerity[i]}[discharges[i]>reach.qstagnant] for i in range(len(discharges)))
+        
+        # check if all celerities are equal to 0. If so, the impact would be 0.
```

Comment:
should this be part of celerity or of configuration initializer?

Reply:
good idea, will move to celerity base class

Reply:
Still good idea, but i like to do this in DFAST-130